ldap/servers/slapd/back-ldbm/db-mdb/mdb_layer.c

```diff
@@ -2338,17 +2338,17 @@ int dbmdb_cursor_get_recno(dbi_cursor_t *cursor, MDB_val *dbmdb_key, MDB_val *db
         rc = MDB_CURSOR_OPEN(mdb_cursor_txn(cursor->cur), mdb_cursor_dbi(cursor->cur), &newcur);
     }
     if (rc == 0) {
-        rc = MDB_CURSOR_GET(newcur, &rce->key, &rce-> data, MDB_SET);
+        rc = MDB_CURSOR_GET(newcur, &rce->key, &rce->data, MDB_SET);
     }
     while (rc == 0) {
         cmpres = dbmdb_cmp_dbi_record(mdb_cursor_dbi(cursor->cur), &curpos_key, &curpos_data, &rce->key, &rce->data);
-        if (cmpres >= 0) {
+        if (cmpres <= 0) {
```

Comment:
To be honest I read and reread this code several times and still do not understand it.
I understand that rce starts from the beginning of index and count recno up to curpos_key.
I wound assume that rce.key is lower than cursor_key during the loop and that we stop the loop when it gets higher or equal (the previous code).

The new code is working even if I do not understand it ;).
I would appreciate comments or examples in the code to explain the loop.

Reply:
> I understand that rce starts from the beginning of index and count recno up to curpos_key.
> I wound assume that rce.key is lower than cursor_key during the loop and that we stop the loop when it gets higher or equal
You are right up to that point: the loop should go on while  cursor_key > rce.key
and should stop it when cursor_key <= rce.key
> (the previous code).
That is wrong: cursor_key <= rce.key is cmpres <= 0 (the new code)
( Looks like I am not the only one that inverts the test ! :wink: )